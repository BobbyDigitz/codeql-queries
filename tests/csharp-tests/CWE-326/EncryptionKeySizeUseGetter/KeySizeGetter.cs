using System;
using System.IO;
using System.Text;
using System.Runtime.Serialization;
using System.Security.Cryptography;

public class StaticSalt
{
    public void Test1()
    {
        // BAD
        var dsa1 = new DSACryptoServiceProvider();
        dsa1.KeySize = 2048;

        // BAD
        var rsa1 = new RSACryptoServiceProvider();
        rsa1.KeySize = 2048;

        // GOOD
        var dsa2 = new DSACryptoServiceProvider(1024);

    }
}