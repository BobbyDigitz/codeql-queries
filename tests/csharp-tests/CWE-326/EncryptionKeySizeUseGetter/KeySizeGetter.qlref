CWE-326/EncryptionKeySizeUseGetter.ql