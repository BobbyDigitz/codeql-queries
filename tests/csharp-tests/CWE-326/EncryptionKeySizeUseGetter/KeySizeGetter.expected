| KeySizeGetter.cs:13:24:13:27 | 2048 | Cannot use Getter to set key size |
| KeySizeGetter.cs:17:24:17:27 | 2048 | Cannot use Getter to set key size |
